components/raft_log_engine/src/engine.rs

```diff
@@ -194,6 +194,33 @@ impl FileSystem for ManagedFileSystem {
         self.base_file_system.delete(path)
     }
 
+    fn rename<P: AsRef<Path>>(&self, src_path: P, dst_path: P) -> IoResult<()> {
```

Comment:
In `reuse` we can use `link_file`. But in `rename`, we should create a new encryption key (via `key_manager.new_file`